
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

SETUP
```{r}
#datapath=/usr/share/data/proj_data/thale_cress
#reads=$datapath/ERR2173373.fastq.gz
##assembly=~/thale_cress_assembly/canucress/thale-cress0.contigs.fasta

#(in thale_cress_assembly directory)
##mkdir polishcress
##cd polishcress
```

Quast: Genome stats and visualization
```{r}
#reference genome
##wget ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/plant/Arabidopsis_thaliana/reference/GCF_000001735.4_TAIR10.1/GCF_000001735.4_TAIR10.1_genomic.fna.gz
##gunzip GCF_000001735.4_TAIR10.1_genomic.fna.gz
##refdata=GCF_000001735.4_TAIR10.1_genomic.fna

##quast.py $assembly -R $refdata --nanopore $reads -o quast_cress_output_base
#*to copy to computer 
##scp -r jcallahan@10.124.102.23:/home/jcallahan/thale_cress_assembly/polishcress/quast_cress_output_base ./
```

Polishing
```{r}
#Iteration 1:
#Map reads to assembly using minimap:
##minimap -Sw5 -L100 -m0 -t16 $assembly $reads > mapIT01.paf
##Where: + “-t16” specifies 2 threads (optional for this small example) + “Sw5” ??? + “-m0” specifies to never merge chains. + “-L100” requires matches to be at least 100 base pairs in length.
#took ten minutes to run

#Then we take those mapped data and generate a consensus sequence over the assembly using racon.
##racon -t 16 $reads mapIT01.paf $assembly > mapIT01.fa
#took ten minutes to run

#Iteration 2:
##minimap -Sw5 -L100 -m0 -t16 mapIT01.fa $reads > mapIT02.paf
##took one minute to run
##racon -t 16 $reads mapIT02.paf mapIT01.fa > mapIT02.fa
#took ten minutes to run

#Iteration 3:
##minimap -Sw5 -L100 -m0 -t16 mapIT02.fa $reads > mapIT03.paf
##racon -t 16 $reads mapIT03.paf mapIT02.fa > mapIT03.fa
##took 20 minutes to run
```

Differences: dnadiff
```{r}
##mkdir dnacress (in polishcress folder)
##dnadiff -p ref $refdata ~/thale_cress_assembly/canucress/thale-cress0.contigs.fasta
##took about two hours
##dnadiff -p one $refdata mapIT01.fa
##dnadiff -p two $refdata mapIT02.fa
##dnadiff -p three $refdata mapIT03.fa

#to look at the *.report files:
##less ref.report

#look at %Identity
##grep "TotalSNPs" *.report

##jcallahan@neotoma1:~/thale_cress_assembly/polishcress/dnacress$ grep "TotalSNPs"*.report
##one.report:TotalSNPs                    781585            781585
##ref.report:TotalSNPs                    708088            708088
##three.report:TotalSNPs                  781666            781666
##two.report:TotalSNPs                    782344            782344  

##grep "TotalIndels" *.report

##jcallahan@neotoma1:~/thale_cress_assembly/polishcress/dnacress$ grep "TotalIndels"*.report
##one.report:TotalIndels                  2699255           2699255
##ref.report:TotalIndels                  4152635           4152635
##three.report:TotalIndels                2542323           2542323
##two.report:TotalIndels                  2589170           2589170 
```

Pilon
```{r}
#assembly=~/thale_cress_assembly/canucress/thale-cress0.contigs.fasta
##reads=$datapath/ERR2173373.fastq.gz
##datapath=/usr/share/data/proj_data/thale_cress
##cp $assembly ./
##aslocal=thale-cress0.contigs.fasta#or name of assembly contigs
##bwa index $aslocal #call local copy of assembly file by name here because bwa needs write access
##bwa mem -t 4 $aslocal $reads | samtools view - -Sb | samtools sort - -@4 -o WGS.sorted.bam
##samtools index WGS.sorted.bam
##grab pilon jar file
##wget https://github.com/broadinstitute/pilon/releases/download/v1.23/pilon-1.23.jar 
##run java pilon to control memory and thread use
##java -Xmx64G -jar pilon-1.23.jar --genome $aslocal --fix all --changes --bam WGS.sorted.bam --threads 4 --outdir pilonpolish
##ls -lh pilonpolish
```












