---
title: "Genome Assembling"
author: "Prof. Harbert"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document:
    theme: united
    highlight: tango
    df_print: paged
    toc: true
    toc_float: true
    toc_depth: 4
  
---


```{r setup, include=FALSE}
library(reticulate)
use_python('/path/to/python') #EDIT
knitr::opts_chunk$set(echo = TRUE)
```

[home](https://rsh249.github.io/applied_bioinformatics)



# Genome Assembly: Canu

[Canu paper](https://genome.cshlp.org/content/27/5/722.full.pdf)

[What is N50?](http://www.molecularecologist.com/2017/03/whats-n50/)

Tutorial adapted from Canu Quick Start Guide: https://canu.readthedocs.io/en/latest/quick-start.html

*DO NOT run*

1) Get tutorial nanopore data for E. coli

```{bash, eval=F}
curl -L -o oxford.fasta http://nanopore.s3.climb.ac.uk/MAP006-PCR-1_2D_pass.fasta
```

2) Run Canu

```{bash, eval=F}
canu -p ecoli -d ecoli-oxford  genomeSize=4.8m maxThreads=8  -nanopore-raw oxford.fasta
```

## Examining results

Most of what you will be interested in the *contigs* files in the ecoli-oxford folder. 

Try: 

```{bash, eval=F}
 grep ">" ecoli-oxford/ecoli.contigs.fasta
```

How many contigs did we find and how big are they?

### Explore Canu for your samples

Between now and Monday attempt a Canu run on your data.

[home](https://rsh249.github.io/applied_bioinformatics)



