---
title: "Nanopore Data Workshop"
author: "Prof. Harbert"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document:
    theme: united
    highlight: tango
    df_print: paged
    toc: true
    toc_float: true
    toc_depth: 4
  
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

[home](https://rsh249.github.io/applied_bioinformatics)


# Filter reads by mapping to a reference genome:

In this example we map raw reads to a chloroplast genome and keep only the reads that map. This should make it possible to perform a chloroplast-only assembly.

Useful links:

[Sorting out mapped and unmapped reads with samtools](https://www.biostars.org/p/56246/)

[NCBI genomes ftp](ftp://ftp.ncbi.nih.gov/genomes/)

```{bash, eval=F}
assembly="~/oryza_project/39947_ref_IRGSP-1.0_chrPltd.fa""
reads=/usr/share/data/proj_data/oryza/merge_data/final.fastq
cp $assembly ./
aslocal=39947_ref_IRGSP-1.0_chrPltd.fa

bwa index $aslocal 
bwa mem -t 48 $aslocal $reads | samtools view -F 4 -Sb | samtools sort - -@4 -o WGS.sorted.bam

samtools index WGS.sorted.bam
samtools fastq WGS.sorted.bam > output.fastq 

```

Then run canu as normal:

```{bash, eval=F}
canu -p plast -d plast_canu genomeSize=1m maxThreads=8  -nanopore-raw output.fastq
```


[home](https://rsh249.github.io/applied_bioinformatics)