<!DOCTYPE html>
<html>
<head>
  <title>Sequence Data - File Formats and Major Repositories</title>
  <meta charset="utf-8">
  <meta name="description" content="Sequence Data - File Formats and Major Repositories">
  <meta name="author" content="Prof. Harbert">
  <meta name="generator" content="slidify" />
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/default.css" media="all" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/phone.css" 
    media="only screen and (max-device-width: 480px)" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/slidify.css" >
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/dark.css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->  <link rel=stylesheet href="libraries/widgets/bootstrap/css/bootstrap.css"></link>

  
  <!-- Grab CDN jQuery, fall back to local if offline -->
  <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.7.min.js"></script>
  <script>window.jQuery || document.write('<script src="libraries/widgets/quiz/js/jquery.js"><\/script>')</script> 
  <script data-main="libraries/frameworks/io2012/js/slides" 
    src="libraries/frameworks/io2012/js/require-1.0.8.min.js">
  </script>
  
  

</head>
<body style="opacity: 0">
  <slides class="layout-widescreen">
    
    <!-- LOGO SLIDE -->
        <slide class="title-slide segue nobackground">
  <hgroup class="auto-fadein">
    <h1>Sequence Data - File Formats and Major Repositories</h1>
    <h2>BIO332 Applied Bioinformatics</h2>
    <p>Prof. Harbert<br/></p>
  </hgroup>
  <article></article>  
</slide>
    

    <!-- SLIDES -->
    <slide class="segue dark nobackground" id="slide-1" style="background:;">
  <article data-timings="">
    <p class='.</p>

 '></article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-2" style="background:;">
  <hgroup>
    <h2>Goals</h2>
  </hgroup>
  <article data-timings="">
    <p>Today we will:</p>

<ul>
<li>View and discuss several file formats for storing and distributing DNA sequence data.</li>
<li>Visit and interact with some of the major repositories of sequence data</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-3" style="background:;">
  <hgroup>
    <h2>But First ...</h2>
  </hgroup>
  <article data-timings="">
    <div align="center">
<iframe width="560" height="315" src="https://www.youtube.com/embed/Eul1NlwytF4" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</div>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-4" style="background:;">
  <hgroup>
    <h2>Sequence Files: FASTA</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>The oldest file format currently in use and the default file for many applications.</li>
<li>Stores DNA, RNA, or Amino Acid sequence data only.</li>
</ul>

<p>Consists of two parts:</p>

<ul>
<li>A header line starting with &quot;&gt;&quot;, typically less than 80 characters but format is open.</li>
<li>Sequence data on one or more lines.</li>
<li>Multiple entries may be separated by two newlines.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-5" style="background:;">
  <hgroup>
    <h2>FASTA</h2>
  </hgroup>
  <article data-timings="">
    <p>Adapted from: <a href="https://zhanglab.ccmb.med.umich.edu/FASTA/">https://zhanglab.ccmb.med.umich.edu/FASTA/</a></p>

<p>Stands for &quot;FAST-All&quot;</p>

<p>Files usually end in &#39;.fa&#39; or &#39;.fna&#39;</p>

<p>Looks like:</p>

<pre><code class="bash">&gt;gi|186681228|ref|YP_001864424.1| phycoerythrobilin:ferredoxin oxidoreductase
MNSERSDVTLYQPFLDYAIAYMRSRLDLEPYPIPTGFESNSAVVGKGKNQEEVVTTSYAFQTAKLRQIRA
</code></pre>

<p>The header line typically contains some metadata about the sequence: What it is, where it came from, and any ID numbers associated with that sequence.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-6" style="background:;">
  <hgroup>
    <h2>Nucleic Acid Coding</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="bash">        A --&gt; adenosine           M --&gt; A C (amino)
        C --&gt; cytidine            S --&gt; G C (strong)
        G --&gt; guanine             W --&gt; A T (weak)
        T --&gt; thymidine           B --&gt; G T C
        U --&gt; uridine             D --&gt; G A T
        R --&gt; G A (purine)        H --&gt; A C T
        Y --&gt; T C (pyrimidine)    V --&gt; G C A
        K --&gt; G T (keto)          N --&gt; A G C T (any)
                                  -  gap of indeterminate length
</code></pre>

<p>Normally you will only see A,C,G,T, N (undetermined) and - (gap). </p>

<p>Rarely R/Y Coding is used to simplify some analyses. </p>

<p>Other ambiguous base characters are almost never used, but are accepted convention.</p>

<p>Gap characters (-) are usually assumed to represent single base, but notice that this is coded as &quot;Gap of indeterminate lenght&quot;. So it is NOT always safe to assume (-) represents a single missing base.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-7" style="background:;">
  <hgroup>
    <h2>Amino Acid Coding</h2>
  </hgroup>
  <article data-timings="">
    <p>Though we are unlikely to see amino acid sequences in this class you should be able to recognize when the 20 letter AA code is being used.</p>

<pre><code class="bash">  Letter Code Name
    A ALA alanine                         P PRO proline
    B ASX aspartate or asparagine         Q GLN glutamine
    C CYS cystine                         R ARG arginine
    D ASP aspartate                       S SER serine
    E GLU glutamate                       T THR threonine
    F PHE phenylalanine                   U     selenocysteine
    G GLY glycine                         V VAL valine
    H HIS histidine                       W TRP tryptophan
    I ILE isoleucine                      Y TYR tyrosine
    K LYS lysine                          Z GLX glutamate or glutamine
    L LEU leucine                         X     any
    M MET methionine                      *     translation stop
    N ASN asparagine                      -     gap of indeterminate length
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-8" style="background:;">
  <hgroup>
    <h2>fastq</h2>
  </hgroup>
  <article data-timings="">
    <p>A relative of the FASTA file format is the fastq, where &#39;q&#39; stands for quality. </p>

<p>fastq is a more complex and information rich file format that became necessary when people figured out how to measure base quality on raw sequence data. <a href="https://en.wikipedia.org/wiki/Phred_quality_score">https://en.wikipedia.org/wiki/Phred_quality_score</a></p>

<p>Files end in &#39;.fq&#39; or &#39;.fastq&#39;</p>

<p>The format consists of 4 lines:</p>

<pre><code class="bash">

@SIM:1:FCX:1:15:6329:1045 1:N:0:2
TCGCACTCAACGCCCTGCATATGACAAGACAGAATC
+
&lt;&gt;;##=&gt;&lt;9=AAAAAAAAAA9#:&lt;#&lt;;&lt;&lt;&lt;????#=

</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-9" style="background:;">
  <hgroup>
    <h2>fastq (cont.)</h2>
  </hgroup>
  <article data-timings="">
    <p>The header line always starts with an &#39;@&#39; and can contain information about the sequencing protocol. Usually ends in a read number or ID.</p>

<p>Line 2 is always the sequence itself using the same coding as before.</p>

<p>Line 3 is always a &#39;+&#39;</p>

<p>Line 4 is the quality string where each character encodes a quality score for each base in the sequence.</p>

<p><a href="http://support.illumina.com/content/dam/illumina-support/help/BaseSpaceHelp_v2/Content/Vault/Informatics/Sequencing_Analysis/BS/swSEQ_mBS_FASTQFiles.htm">More info here</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-10" style="background:;">
  <hgroup>
    <h2>Quality scores: Phred</h2>
  </hgroup>
  <article data-timings="">
    <p>fastq is the default file format for all Illumina sequencing. The default quality scoring for modern Illumina platforms is known as Phred33.</p>

<p>This is what all those symbols mean in Phred:</p>

<p><a href="http://www.drive5.com/usearch/manual7/quality_score.html">
<img src="./images/qscores.gif" title="plot of chunk unnamed-chunk-5" alt="plot of chunk unnamed-chunk-5" width="70%" style="display: block; margin: auto;" />
</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-11" style="background:;">
  <hgroup>
    <h2>Visualizing Quality</h2>
  </hgroup>
  <article data-timings="">
    <p>For Illumina and Sanger sequencing the common program to analyze DNA sequence read quality is called FastQC</p>

<p><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-12" style="background:;">
  <hgroup>
    <h2>Nanopore: Fast5</h2>
  </hgroup>
  <article data-timings="">
    <p>ONT applications all use and create Fast5 files.</p>

<p>Fast 5 is an application of the <a href="https://support.hdfgroup.org/HDF5/doc/Specs.html">HDF5 standard</a></p>

<p>These are &#39;binary&#39; database files that require special software tools to read the data. This means two things:</p>

<ul>
<li><p>Fast5 is an information rich and flexible format that provides an efficient data/metadata storage platform.</p></li>
<li><p>we cannot &quot;see&quot; the Fast5 files directly (i.e., using &#39;cat&#39; or &#39;nano&#39;)</p></li>
</ul>

<p>We will look at some of the tools for viewing and extracting data from Fast5 in a later session.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-13" style="background:;">
  <hgroup>
    <h2>Major Sequence Data Archives</h2>
  </hgroup>
  <article data-timings="">
    <p>There are two places that you should go to if you want to find published sequence data: (NOTE: We will be searching these for reference data for our Nanopore projects)</p>

<p>1) The US NCBI databases: <a href="https://www.ncbi.nlm.nih.gov/">https://www.ncbi.nlm.nih.gov/</a> (includes Genbank)</p>

<p>2) The European EMBL databases: <a href="https://www.ebi.ac.uk/">https://www.ebi.ac.uk/</a></p>

<p>What kind of files do we recover from these searches?</p>

<p>(Hint: Fast5 <a href="https://www.ebi.ac.uk/ena/data/view/PRJEB20906">here</a>)</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-14" style="background:;">
  <hgroup>
    <h2>Next: Nanopore Community</h2>
  </hgroup>
  <article data-timings="">
    <p>You should all have received and email to join my Nanopore Community group.</p>

<p><a href="https://community.nanoporetech.com/">https://community.nanoporetech.com/</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-15" style="background:;">
  <article data-timings="">
    
  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="segue" id="slide-16" style="background:;">
  <hgroup>
    <h1>Questions?</h1>
  </hgroup>
  <article data-timings="">
    <p><a href="https://rsh249.github.io/applied_bioinformatics">home</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-17" style="background:;">
  <article data-timings="">
    
  </article>
  <!-- Presenter Notes -->
</slide>

    <slide class="backdrop"></slide>
  </slides>
  <div class="pagination pagination-small" id='io2012-ptoc' style="display:none;">
    <ul>
      <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=1 title='NA'>
         1
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=2 title='Goals'>
         2
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=3 title='But First ...'>
         3
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=4 title='Sequence Files: FASTA'>
         4
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=5 title='FASTA'>
         5
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=6 title='Nucleic Acid Coding'>
         6
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=7 title='Amino Acid Coding'>
         7
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=8 title='fastq'>
         8
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=9 title='fastq (cont.)'>
         9
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=10 title='Quality scores: Phred'>
         10
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=11 title='Visualizing Quality'>
         11
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=12 title='Nanopore: Fast5'>
         12
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=13 title='Major Sequence Data Archives'>
         13
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=14 title='Next: Nanopore Community'>
         14
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=15 title='NA'>
         15
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=16 title='Questions?'>
         16
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=17 title='NA'>
         17
      </a>
    </li>
  </ul>
  </div>  <!--[if IE]>
    <script 
      src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
    </script>
    <script>CFInstall.check({mode: 'overlay'});</script>
  <![endif]-->
</body>
  <!-- Load Javascripts for Widgets -->
  <script src="libraries/widgets/bootstrap/js/bootstrap.min.js"></script>
<script src="libraries/widgets/bootstrap/js/bootbox.min.js"></script>
<script src="http://ajax.googleapis.com/ajax/libs/jqueryui/1.10.3/jquery-ui.min.js"></script>
<script src="http://bartaz.github.io/sandbox.js/jquery.highlight.js"></script>

  <script>  
  $(function (){ 
    $("#example").popover(); 
    $("[rel='tooltip']").tooltip(); 
  });  
  </script>  
  <!-- LOAD HIGHLIGHTER JS FILES -->
  <script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
  <!-- DONE LOADING HIGHLIGHTER JS FILES -->
   
  </html>